crates/gitbutler-repo/src/commands.rs

```diff
@@ -60,12 +61,47 @@ impl RepoCommands for Project {
             let entry = tree.get_path(relative_path)?;
             let blob = ctx.repository().find_blob(entry.id())?;
```

Comment:
I am aware that these semantics were present already, but I am confused that the lookup is in the tree first. My expectation would be the following order:

* worktree
* index
* `HEAD^{tree}`

Reply:
That’s why it seems that untracked binary files cannot be previewed.

Reply:
Just now when working on #5111 I also touched the method, and a 'blame' later I found out that I think it's best to ask @ndom91 what should be done here. Thanks in advance for your help.

Reply:
Yeah I wrote the initial get repo content for a specific use case, it would need to be modified and/or built on top of for this. 

I love the motivation though! Happy to pick this up with you on Monday 👌